### DISCUSSION TITLE
Host docs here instead of ctsm-docs?

### DISCUSSION THREAD

**Discussion by samsrabin** [ESCOMP/CTSM#2819] [Ideas]
CIME is able to [generate documentation previews on PRs](https://github.com/ESMCI/cime/blob/62a9b1747272b586521d9f22d29977733edacfcb/.github/workflows/docs.yml#L90) and [automatically update their documentation upon pushes to master](https://github.com/ESMCI/cime/blob/62a9b1747272b586521d9f22d29977733edacfcb/.github/workflows/docs.yml#L124) because they host their docs in the same repo as their code. Our use of the separate ctsm-docs repo makes this impossible (or at least difficult enough that I can't figure it out given my lack of Github workflow experience).

What was the justification for having the separate ctsm-docs repo again? Is it still justified?

CC @billsacks

**Comment by billsacks** [comment]
Here's what I wrote in an email on this in Aug, 2020; I haven't checked how relevant / correct this still is:

> The rationale for having a separate repository for the documentation build is: If this were in the main repo, people would get these builds whenever they clone the repo. For the CTSM documentation, even a single build doubles or triples the size of the repository. As we start supporting more side-by-side builds, the issue of repository bloat will grow. Since most people won't want or need these doc builds, I thought it would be better to isolate these builds to their own repository rather than having them on an orphan branch in the main repository. For CISM, this isn't as much of an issue, but since I was setting up the CISM documentation around the same time as the CTSM documentation, I set them up the same way for consistency.

**Reply by samsrabin** [reply]
That's a good point, Bill, thanks for the context. However, there's now a way to publish that seemingly doesn't have that downside: [with GitHub Actions](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow). Importantly, it [doesn't use a build branch](https://github.blog/news-insights/product-news/github-pages-now-uses-actions-by-default/#:~:text=A%20source%20branch%20is%20not%20required%20anymore), so there's nothing to see in a clone.

**Reply by billsacks** [reply]
Oh, interesting - good to know. So this new method doesn't actually store the build results in the repository? Interesting.

**Reply by billsacks** [reply]
In that case, I agree that your plan makes sense. Thank you for all of your work on this!

**Comment by wwieder** [comment]
Is this discussion thread still relevant given ongoing work to improve our documentation process being led by @samsrabin ?
